Enable dynamically ciflow/slow so that we can run GHA slow tests on PR

zhouzhuojie opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #61987



This PR enables us to run slow GHA tests on PR.

Steps to do (~may only take effect after this PR is merged~ works on this PR)
- Add label `ciflow/slow`
- Assign/unassign pytorchbot
- The job should be running .github/workflows/pytorch-linux-xenial-cuda10.2-cudnn7-py3.6-gcc7.yml

The above steps are manual, and after probot can do the dispatch work, the ciflow will be automated.

Related meta RFC issue: #61888

Differential Revision: [D29832758](https://our.internmc.facebook.com/intern/diff/D29832758)

zhouzhuojie committed:
Add ciflow/slow so that we can enable slow tests on PR

[ghstack-poisoned]

zhouzhuojie committed:
Update on "Add ciflow/slow so that we can enable slow tests on PR"




This PR enables us to run slow GHA tests on PR.

Steps to do (may only take effect after this PR is merged)
- Add label `ciflow/slow`
- Assign/unassign pytorchbot
- The job should be running .github/workflows/pytorch-linux-xenial-cuda10.2-cudnn7-py3.6-gcc7.yml

The above steps are manual, and after probot can do the dispatch work, the ciflow will be automated.

[ghstack-poisoned]

janeyx99 commented on .github/scripts/generate_ci_workflows.py:
```diff
@@ -49,6 +49,7 @@
 class CIFlowConfig:
     enabled: bool = False
     labels: Set[str] = field(default_factory=set)
+    trigger_action_only: bool = False
```
Just the clarify, does trigger_action_only being True mean that the workflow will only run if the trigger action is enabled and not for any other case?

zhouzhuojie replied:
```
    {%- if ciflow_config.trigger_action_only %}
    types: [!{{ ciflow_config.trigger_action }}]
    {%- else %}
    types: [opened, synchronize, reopened, !{{ ciflow_config.trigger_action }}]
    {%- endif %}
```

It controls if we listen on 
- types: `[opened, synchronize, reopened, !{{ ciflow_config.trigger_action }}]` or just 
- types: `[!{{ ciflow_config.trigger_action }}]`

Going to add some comments...

janeyx99 commented:
Hm...I don't see Linux CI: pytorch-linux-xenial-cuda10.2-cudnn7-py3.6-gcc7 in the CI.

zhouzhuojie committed:
Update on "Enable dynamically ciflow/slow so that we can run GHA slow tests on PR"




This PR enables us to run slow GHA tests on PR.

Steps to do (may only take effect after this PR is merged)
- Add label `ciflow/slow`
- Assign/unassign pytorchbot
- The job should be running .github/workflows/pytorch-linux-xenial-cuda10.2-cudnn7-py3.6-gcc7.yml

The above steps are manual, and after probot can do the dispatch work, the ciflow will be automated.

Related meta RFC issue: #61888

[ghstack-poisoned]

zhouzhuojie committed:
Update on "Enable dynamically ciflow/slow so that we can run GHA slow tests on PR"




This PR enables us to run slow GHA tests on PR.

Steps to do (may only take effect after this PR is merged)
- Add label `ciflow/slow`
- Assign/unassign pytorchbot
- The job should be running .github/workflows/pytorch-linux-xenial-cuda10.2-cudnn7-py3.6-gcc7.yml

The above steps are manual, and after probot can do the dispatch work, the ciflow will be automated.

Related meta RFC issue: #61888

[ghstack-poisoned]

zhouzhuojie commented:
> Hm...I don't see Linux CI: pytorch-linux-xenial-cuda10.2-cudnn7-py3.6-gcc7 in the CI.

now it's there, I just unassigned pytorchbot

janeyx99 commented:
> Hm...I don't see Linux CI: pytorch-linux-xenial-cuda10.2-cudnn7-py3.6-gcc7 in the CI.

Oh I see it now! :D

zhouzhuojie commented:
@zhouzhuojie has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29832758).

driazati commented on .github/scripts/generate_ci_workflows.py:
```diff
@@ -228,6 +233,12 @@ def generate_workflow_file(self, workflow_template: jinja2.Template) -> None:
+        ciflow_config=CIFlowConfig(
+            enabled=True,
+            trigger_action_only=True,
+            labels=set(['ciflow/slow']),
```
nit:
```suggestion
            labels={'ciflow/slow'},
```

zhouzhuojie replied:
ha, always forget the set literal, will change in the following PRs
